crates/sui/src/keytool.rs

```diff
@@ -257,6 +281,88 @@ impl KeyToolCommand {
+                let mut hasher = DefaultHash::default();
+                hasher.update(bcs::to_bytes(&intent_msg)?);
+                let digest = hasher.finalize().digest;
+                println!("Digest to sign: {:?}", Base64::encode(digest));
+               
+                // Set up the KMS client, expect to use env vars or whatever the AWS
+                // SDK prefers.
+                // Length of Pubkey - 33, sig - 64
+                
+                let region = Region::default();
+                let kms = KmsClient::new(region);
+
+                // Get Pub key from keyid. This is in DER Format from AWS
+                let req = GetPublicKeyRequest { grant_tokens: None, key_id: keyid.to_string() };
+                let pubkey_resp = kms.get_public_key(req).await.unwrap();
+                let public_key_bytes = pubkey_resp.public_key.unwrap_or_default();
+                let public_key: &[u8] = &public_key_bytes.to_vec(); // Vec of bytes
+
+                // Parse the PEM-encoded public key and extract the raw bytes
+                // Compresses into 33 bytes
+                let group = EcGroup::from_curve_name(Nid::SECP256K1).unwrap();
+                let pkey = PKey::public_key_from_der(public_key).unwrap();
+                let pkey_compact_result = pkey.ec_key().unwrap().public_key().to_bytes(&group,
+                    PointConversionForm::COMPRESSED,
+                    &mut openssl::bn::BigNumContext::new().unwrap(),);
+                let pkey_compact = pkey_compact_result.unwrap();
+
+                // Generates Corresponding Sui Address from public key
+                let secp_pk = Secp256k1PublicKey::from_bytes(&pkey_compact).unwrap();
+                println!(
+                        "Address For Corresponding KMS Key: {}",
+                        Into::<SuiAddress>::into(&secp_pk),
+                );
+                    
+                // Construct the signing request
+                let request = SignRequest {
+                    key_id: keyid.to_string() ,
+                    message: digest.to_vec().into(),
+                    message_type: Some("RAW".to_string()),
+                    signing_algorithm: "ECDSA_SHA_256".to_string(),
+                    ..Default::default()
+                };
+                // Sign the message, normalize the signature and then compacts it
+                let response = kms.sign(request).await.unwrap();
+                let sig_bytes_der = response.signature.map(|b| b.to_vec()).unwrap_or_default();     
+                let mut sig = secpSig::from_der(&sig_bytes_der).unwrap();
+                let _ = secpSig::normalize_s(&mut sig);
+                let sig_bytes = secpSig::serialize_compact(&sig);
+
+
+                //println!("Sig Bytes {:?}", sig_bytes);
+                //println!("Public Key Compact bytes {:?}", pkey_compact);
+            
+                // TODO, just plopping more things into flag
+                let mut flag = vec![0x01];
```

Comment:
prefer constant instead of magic number: SignatureScheme::Secp256k1.flag()

Reply:
makes sense, changed